src/cai_lab_to_nwb/zaki_2024/zaki_2024_convert_session.py

```diff
@@ -144,6 +147,20 @@ def session_to_nwb(
     elif verbose and not sleep_classification_file_path.is_file():
         print(f"No sleep classification output csv file found at {sleep_classification_file_path}")
 
+    # Add Shock Stimuli times for FC sessions
+    if include_shock_times:
+        import re
+
+        subjects_df = pd.read_excel(data_dir_path / "Ca_EEG_Design.xlsx")
```

Comment:
Mmm interesting.

I would put the shock amplitude decoding code inside the interface because is more complicated and I think adds unnecessary details to the conversion script. On the other hand, the shock-times are just hard-coded and they are probably better outside.

If this format of the amplitudes does not look very standard (maybe Joe prepare it just for this conversion) I would just hard-write the values here. It would be easier by someone wanting to add shock values to modify them (they are few) and they don't need the parsing code if it was a one-off thing

What do you think?

Reply:
You're right! It would be probably even better to pass directly a dictionary with all info about shock stimuli, directly to the session_to_nwb function so that I dn't need the include_shock_times flag